docs: Fix some issues in the user guide

corycaywood opened the pull request:
# Description
The MiniApp#create function was still using the old method (which has now been removed). Also removed the note about AppCompat because this was relevant to Android 22 and below only (we now support 23+).


# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

minh-rakuten reviewed (commented):
Thank you for updating Userguide.

`Note: This SDK uses suspend functions, so you should use Kotlin Coroutines when calling the functions. These examples use Dispatchers.Default, but you can use whichever CoroutineContext and CouroutineScope that is appropriate for your App as long as it does not run in the main thread.`
-> I'm sorry that I miss this in document before. We should also specify that the Miniapp api call should not be executed in main thread (UI).

corycaywood commented:
> Thank you for updating Userguide.
> 
> `Note: This SDK uses suspend functions, so you should use Kotlin Coroutines when calling the functions. These examples use Dispatchers.Default, but you can use whichever CoroutineContext and CouroutineScope that is appropriate for your App as long as it does not run in the main thread.`
> -> I'm sorry that I miss this in document before. We should also specify that the Miniapp api call should not be executed in main thread (UI).

Yeah, good catch. I've updated this and also added this to the `TroubleShooting` section.

Also, I updated the `MiniAppSdkException` so that it will set a custom message when `NetworkOnMainThreadException` is thrown because that exception has a `null` message.

nitish-a-jain commented on miniapp/USERGUIDE.md:
```diff
@@ -37,9 +37,9 @@ The SDK is configured via manifest meta-data, the configurable values are:
-| Host App User Agent Info     | String  | `com.rakuten.tech.mobile.miniapp.HostAppUserAgentInfo` | ✅         | 🚫        |
 | App ID                       | String  | `com.rakuten.tech.mobile.ras.AppId`                    | ❌         | 🚫        |
 | RAS Project Subscription Key | String  | `com.rakuten.tech.mobile.ras.ProjectSubscriptionKey`   | ❌         | 🚫        |
+| Host App User Agent Info     | String  | `com.rakuten.tech.mobile.miniapp.HostAppUserAgentInfo` | ✅         | 🚫        |
```
I think Host App Version can also be grouped downwards in the list.
```suggestion
| Host App Version             | String  | `com.rakuten.tech.mobile.miniapp.HostAppVersion`       | ❌         | 🚫        |
| Host App User Agent Info     | String  | `com.rakuten.tech.mobile.miniapp.HostAppUserAgentInfo` | ✅         | 🚫        |
```

nitish-a-jain commented on miniapp/USERGUIDE.md:
```diff
@@ -37,9 +37,9 @@ The SDK is configured via manifest meta-data, the configurable values are:
 |------------------------------|---------|--------------------------------------------------------|----------- |--------- |
 | Base URL                     | String  | `com.rakuten.tech.mobile.miniapp.BaseUrl`              | ❌         | 🚫        |
 | Host App Version             | String  | `com.rakuten.tech.mobile.miniapp.HostAppVersion`       | ❌         | 🚫        |
```
```suggestion
```

nitish-a-jain commented on miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppSdkException.kt:
```diff
@@ -29,3 +35,9 @@ internal fun sdkExceptionForInvalidArguments(message: String = "") =
     MiniAppSdkException("Only accept context of type Activity or ActivityCompat")
+
+private fun exceptionMessage(exception: Exception) = when (exception) {
+    is NetworkOnMainThreadException -> "Network requests must not be performed on the main thread. " +
```
How about template string or `.plus`?

nitish-a-jain reviewed (approved):
👍 
Just minor comments for consideration.

minh-rakuten reviewed (approved):
Looks good.
